biogrid/apps/api/src/app/services/Biogrid.ts

```diff
@@ -67,25 +67,32 @@ export async function simulateNewBiogrid(
     numberOfLargeBatteryCells: body.largeBatteryCells,
     numberOfSmallBatteryCells: body.smallBatteryCells,
     numberOfSolarPanels: body.numSolarPanels,
+    startDate: body.startDate,
   });
   const biobrain = BioBrain.Instance;
   const initState = biogrid.getSystemState();
   const statesJson = [];
   const currentDate = body.startDate;
+  const efficiencies: number[] = [];
   for (let i = 0; i < constants.simulation.NUMBER_OF_SIM_HOURS; i++) {
     // Start at midnight, increment hours until NUMBER_OF_SIM_HOURS reached
     currentDate.setHours(i);
     biogrid.updateEnergyUsage(currentDate);
     const action = await biobrain.computeAction(initState, currentDate);
     biogrid.takeAction(action);
     statesJson.push(biogrid.getJsonGraphDetails());
+    efficiencies.push(biogrid.getEfficiency());
   }
+  const efficienciesAddedUp = efficiencies
+    .filter((efficiency) => efficiency)
```

Comment:
does this filter do anything? looks like you're just passing through the same element

Reply:
It filters for any NaN values. Added a comment